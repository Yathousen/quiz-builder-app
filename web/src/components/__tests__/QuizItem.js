import React from 'react';
import renderer from 'react-test-renderer';
import { QuizItem } from '../';
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('QuizItem', () => {
  expect(
    renderer
      .create(
        <QuizItem
          data={{
            id: 1,
            key: '',
            name: 'Test 1',
            published: false,
            createdAt: { toDate: () => new Date('2022-04-20') },
            questions: [
              {
                name: 'Question Title',
                types: [
                  { name: 'Single correct answer', selected: true },
                  { name: 'Select all correct answers', selected: false },
                ],
                options: [
                  { name: 'Answer 1', selected: true },
                  { name: 'Answer 2', selected: false },
                ],
              },
            ],
          }}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

it('QuizItem Published', () => {
  expect(
    renderer
      .create(
        <QuizItem
          data={{
            id: 1,
            key: '',
            name: 'Test 1',
            published: true,
            createdAt: { toDate: () => new Date('2022-04-20') },
            questions: [
              {
                name: 'Question Title',
                types: [
                  { name: 'Single correct answer', selected: true },
                  { name: 'Select all correct answers', selected: false },
                ],
                options: [
                  { name: 'Answer 1', selected: true },
                  { name: 'Answer 2', selected: false },
                ],
              },
            ],
          }}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
