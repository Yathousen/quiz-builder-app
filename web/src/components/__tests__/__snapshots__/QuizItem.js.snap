// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizItem 1`] = `
<div
  className="mt-4 px-4 py-4 mx-auto bg-white rounded-lg shadow-md dark:bg-gray-800"
>
  <div
    className="flex items-center justify-between"
  >
    <a
      className="text-2xl font-bold text-gray-700 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 hover:underline cursor-pointer"
      onClick={[Function]}
    >
      Test 1
    </a>
  </div>
  <div
    className="mt-2"
  >
    <p
      className="mt-2 text-gray-600 dark:text-gray-300"
    >
      1 question
    </p>
  </div>
  <span
    className="text-sm font-light text-gray-600 dark:text-gray-400"
  >
    Tuesday, April 19, 2022, 8:00 PM
  </span>
</div>
`;

exports[`QuizItem Published 1`] = `
<div
  className="mt-4 px-4 py-4 mx-auto bg-white rounded-lg shadow-md dark:bg-gray-800"
>
  <div
    className="flex items-center justify-between"
  >
    <a
      className="text-2xl font-bold text-gray-700 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 hover:underline cursor-pointer"
      onClick={[Function]}
    >
      Test 1
    </a>
    <a
      className="px-3 py-1 text-sm font-bold text-gray-100 transition-colors duration-200 transform bg-gray-600 rounded cursor-pointer hover:bg-gray-500"
      onClick={[Function]}
    >
      Published
    </a>
  </div>
  <div
    className="mt-2"
  >
    <p
      className="mt-2 text-gray-600 dark:text-gray-300"
    >
      1 question
    </p>
  </div>
  <span
    className="text-sm font-light text-gray-600 dark:text-gray-400"
  >
    Tuesday, April 19, 2022, 8:00 PM
  </span>
</div>
`;
